query_optimizer/ExecutionGenerator.cpp

```diff
@@ -1635,5 +1635,11 @@ void ExecutionGenerator::convertTableGenerator(
   temporary_relation_info_vec_.emplace_back(tablegen_index, output_relation);
 }
 
+S::InsertDestination* ExecutionGenerator::addNewInsertDestinationToQueryContext() {
+  S::InsertDestination *insert_destination_proto(query_context_proto_->add_insert_destinations());
+  insert_destination_proto->set_query_id(query_context_proto_->query_id());
```

Comment:
I have an alternative implementation suggestions:

 1. Reset changes in this file and `InsertDestination.proto`.
 1. Modify `InsertDestination::ReconstructFromProto` to take a new argument `query_id` from `QueryContext proto`'s field `query_id`.